content/en/docs/reference/configuration/jwk/_index.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "JSON Web Keys (JWK)"
+linkTitle: "JSON Web Keys (JWK)"
+weight: 7
+---
+  
+  
+JSON Web Key (JWK) is a JSON data structure that represents a single public key or a set of
+public keys. IAM uses private JWKs to sign tokens, and IAM clients use public JWKs to validate the tokens.
+A [JWK Generator][jwk-generator] can be used to generate new keys.
+
+### Keystore Location
+
+JWKs are stored in a file that is called keystore. All the keys in this file will be 
+used for validating the tokens. Keystore location is set using the `IAM_KEY_STORE_LOCATION` 
+environment variable, e.g.:
+
+```env
+IAM_KEY_STORE_LOCATION=file:///secrets/jwk/keystore.jwks
+```
+
+
+### Default Key ID (kid)
+
+If there are multiple JWKs in the keystore, `IAM_JWK_DEFAULT_KEY` environment variable
+can be set to choose the the key that signs the tokens, e.g.:
+
+```env
+IAM_JWK_DEFAULT_KEY=rsa2
+```
+
+If this variable is not set it will use the default value `rsa1`.
+
+
+### Default JWE Keys
+
+JSON Web Encryption (JWE) represents encrypted content using JSON-based data structures.
+JWE allows to encrypt the the JWT tokens so that only the intended receipient can decrypt
+and see the content.  
+
+Default JWE decrypt key and encrypt key id can be set using the
+ `IAM_JWK_DEFAULT_JWE_DECRYPT_KEY_ID` and `IAM_JWK_DEFAULT_JWE_ENCRYPT_KEY_ID` 
+ environment variables respectively, e.g.:
+
+ ```env
+IAM_JWK_DEFAULT_JWE_DECRYPT_KEY_ID=jwe-decrypt
+IAM_JWK_DEFAULT_JWE_ENCRYPT_KEY_ID=jwe-encrypt
+```
+
+### Default Algorithms
+
+Default alorithms used for signing and encrypting the key can be set using 
+`IAM_JWK_DEFAULT_JWS_ALGORITHM` and `IAM_JWT_DEFAULT_JWE_ALGORITHM` environment variables
+respectively, default algorithms are like following:
+```env
+IAM_JWK_DEFAULT_JWS_ALGORITHM=RS256
+IAM_JWT_DEFAULT_JWE_ALGORITHM=RSA_OAEP_256
+```
+
+## Rotating Keys
+
+To change the JWKs without any interruption on the service, a rolling update can be done 
+using these configurations. An example of a rolling update can be like this:  
```

Comment:
"An example of a rolling update can be like this:" -> "Below an example of a rolling update."